### DISCUSSION TITLE
Localization Culture Url Change

### DISCUSSION THREAD

**Discussion by burakdmir** [abpframework/abp#17046] [Q&A] [answered]
Hi,

Is there a way in ABP to have the URL in the format '/en' instead of '?culture=en' when we specify 'culture=en' in the URL to handle the language change? I'm asking this because I don't want to disrupt the framework structure and wondered if there is a way to achieve this in ABP or if I should apply the techniques I use in normal ASP.NET Core.

Thanks

**Comment by maliming** [comment, marked as answer]
hi

This won't break the framework. You can add your culture provider.

`culture=en` is coming from asp net core's `QueryStringRequestCultureProvider`.



https://learn.microsoft.com/en-us/aspnet/core/fundamentals/localization/select-language-culture?view=aspnetcore-7.0#configure-localization-middleware

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/localization/select-language-culture?view=aspnetcore-7.0#querystringrequestcultureprovider
